.github/workflows/generate-tag.yml

```diff
@@ -0,0 +1,40 @@
+name: "[Job] Generate tag"
+
+on:
+  workflow_call:
+    outputs:
+      tag:
+        description: "Semver tag of this commit/deployment"
+        value: ${{ jobs.generate_tag.outputs.tag }}
+
+jobs:
+  generate_tag:
+    runs-on: ubuntu-latest
+    outputs:
+      tag: ${{ steps.bump_version.outputs.tag }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: "0"
+      - name: Extract branch name
+        run: |
+          if [ "$GITHUB_EVENT_NAME" == "push" ]; then
```

Comment:
Careful with this it doesn't work with workflow dispatch

Reply:
Switched to GITHUB_REF